nibabel/streamlines/tractogram_file.py

```diff
@@ -0,0 +1,133 @@
+from abc import ABCMeta, abstractmethod
+from nibabel.externals.six import with_metaclass
+
+from .header import Field
+
+
+class ExtensionWarning(Warning):
+    pass
+
+
+class HeaderWarning(Warning):
+    pass
+
+
+class HeaderError(Exception):
+    pass
+
+
+class DataError(Exception):
+    pass
+
+
+class abstractclassmethod(classmethod):
+    __isabstractmethod__ = True
+
+    def __init__(self, callable):
+        callable.__isabstractmethod__ = True
+        super(abstractclassmethod, self).__init__(callable)
+
+
+class TractogramFile(with_metaclass(ABCMeta)):
+    """ Convenience class to encapsulate tractogram file format. """
+
+    def __init__(self, tractogram, header=None):
+        self._tractogram = tractogram
+        self._header = {} if header is None else header
+
+    @property
+    def tractogram(self):
+        return self._tractogram
+
+    @property
+    def streamlines(self):
+        return self.tractogram.streamlines
+
+    @property
+    def header(self):
+        return self._header
+
+    @property
+    def affine(self):
+        return self.header.get(Field.VOXEL_TO_RASMM)
+
+    def get_tractogram(self):
```

Comment:
Some people told me they like `get_` methods. I do prefer to use properties myself and that is why I've also added them. If you tell me NiBabel always properties (or tries too), I will remove those `get_` methods for the sake of consistency.